Routing test fails for feature release branches

**Describe the issue**
Example run: https://github.com/medic/cht-core/actions/runs/26500616089/job/78042547424#step:16:3445

```
  2) routing
       unauthenticated routing
         should display deploy-info to authenticated users:

      AssertionError: expected { application: 'medic', …(10) } to deeply equal { user: '', …(10) }
      + expected - actual

         "time": "2026-05-27T08:45:58.829Z"
         "timestamp": 1779872374456
         "upgrade_log_id": "upgrade_log:1779872369719:5.1.2-FR-attachments-for-subcontacts"
         "user": ""
      -  "version": "5.1.2-FR-attachments-for-subcontacts"
      +  "version": "5.1.2-5.1.2-FR-attachments-for-subcontacts.26500616089-1779871558803"
       }
      
      at /home/runner/work/cht-core/cht-core/tests/integration/api/routing.spec.js:195:42
      at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
```

**Describe the improvement you'd like**
looking at the test output, the test expected `5.1.2-FR-attachments-for-subcontacts` but received `5.1.2-5.1.2-FR-attachments-for-subcontacts.26500616089-1779871558803`. It could be that the `deploy-info` has an invalid shape for feature releases. 

If the test is right, fix the code, if the test is wrong fix the test.

--- a/scripts/build/versions.js
+++ b/scripts/build/versions.js
@@ -10,11 +10,15 @@ const {
   INTERNAL_CONTRIBUTOR,
 } = process.env;
 
+const FEATURE_RELEASE_BRANCH_PATTERN = /^\d+\.\d+\.\d+-FR-.+/;
+
 const escapeBranchName = (branch) => branch?.replace(/[^A-Za-z0-9.-]/g, '-');
+const isFeatureReleaseBranch = (branch) => FEATURE_RELEASE_BRANCH_PATTERN.test(branch);
 
 const getBranchVersion = (release) => {
   const branch = BRANCH === 'master' ? 'alpha' : escapeBranchName(BRANCH);
-  const base = `${packageJson.version}-${branch}`;
+  // Feature-release branches (e.g. 5.1.2-FR-foo) already carry their own version
+  const base = isFeatureReleaseBranch(branch) ? branch : `${packageJson.version}-${branch}`;
   return release ? base : `${base}.${BUILD_NUMBER}`;
 };
 

--- a/tests/integration/api/routing.spec.js
+++ b/tests/integration/api/routing.spec.js
@@ -186,12 +186,16 @@ describe('routing', () => {
         const { BRANCH, TAG } = process.env;
         const isBranchBuild = BRANCH && !TAG;
 
+        // for tags, build_info.version is the tag (semver-valid).
+        // for branches, build_info.version is the escaped branch name: when that itself is
+        // valid semver (e.g. feature-release branches like 5.1.2-FR-foo) the api returns it,
+        // otherwise it falls back to build_info.build (which always carries the build number).
+        const branchVersion = semver.valid(ddoc.build_info.version) || ddoc.build_info.build;
         const deployInfo = {
           ...ddoc.deploy_info,
           ...ddoc.build_info,
-          version: isBranchBuild ? ddoc.build_info.build : ddoc.build_info.version
+          version: isBranchBuild ? branchVersion : ddoc.build_info.version
         };
-        // for historical reasons, for a branch the version in the ddoc is the branch name.
         expect(deployInfoOnline).to.deep.equal(deployInfo);
         expect(deployInfoOffline).to.deep.equal(deployInfo);
       });

